src/common/src/types/datetime.rs

```diff
@@ -111,7 +111,17 @@ type Result<T> = std::result::Result<T, InvalidParamsError>;
 
 impl ToText for Date {
     fn write<W: std::fmt::Write>(&self, f: &mut W) -> std::fmt::Result {
-        write!(f, "{}", self.0)
+        if self.0.year() < 0 {
```

Comment:
Should be `year_ce` here ([chrono rustdoc](https://docs.rs/chrono/0.4.26/chrono/naive/struct.NaiveDate.html#calendar-date)):
> ISO 8601 calendars has the year 0, which is 1 BCE (a year before 1 CE). If you need a typical BCE/BC and CE/AD notation for year numbers, use the [Datelike::year_ce](https://docs.rs/chrono/0.4.26/chrono/trait.Datelike.html#method.year_ce) method.

In PostgreSQL:
```
test=# select to_date('0001-02-03', 'YYYY-MM-DD') - 100; -- minus integer returns date, while minus interval returns timestamp
   ?column?    
---------------
 0001-10-26 BC
(1 row)
```

Other parts of RisingWave are not tested for BC years either. It may be easier to just give users a consistent message that date/time before `1973-03-03 09:46:40` (#8359) are not supported.